### PR TITLE
return devfile v2 as a default devfile

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Return devfile version 2 as a default devfile.

### What issues does this PR fix or reference?

fixes https://github.com/eclipse-che/che/issues/23007

### How to test this PR?

see https://github.com/eclipse-che/che/issues/23007

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/BaseFactoryParameterResolver.java [hunk 1]
```diff
@@ -60,7 +60,12 @@ protected FactoryMetaDto createFactory(
             contentProvider,
             extractOverrideParams(factoryParameters),
             getSkipAuthorisation(factoryParameters))
-        .orElseGet(() -> newDto(FactoryDto.class).withV(CURRENT_VERSION).withSource("repo"))
+        .orElseGet(
+            () ->
+                newDto(FactoryDevfileV2Dto.class)
+                    .withDevfile(Map.of("schemaVersion", "2.3.0"))
+                    .withV(CURRENT_VERSION)
```
Comment (ibuziuk) [inline wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/BaseFactoryParameterResolver.java:67]: what is the CURRENT_VERSION stand for ?

File: wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/BaseFactoryParameterResolver.java [hunk 2]
```diff
@@ -60,7 +60,12 @@ protected FactoryMetaDto createFactory(
             contentProvider,
             extractOverrideParams(factoryParameters),
             getSkipAuthorisation(factoryParameters))
-        .orElseGet(() -> newDto(FactoryDto.class).withV(CURRENT_VERSION).withSource("repo"))
+        .orElseGet(
+            () ->
+                newDto(FactoryDevfileV2Dto.class)
+                    .withDevfile(Map.of("schemaVersion", "2.3.0"))
```
Comment (ibuziuk) [inline wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/BaseFactoryParameterResolver.java:66]: should we extract it to const?

**Review by ibuziuk — COMMENTED** [review]
@vinokurig could you please clarify if we removed everything related to devfile v1? or this is smth. to be done separately?

File: wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/BaseFactoryParameterResolver.java (hunk 1 above)
Comment (vinokurig) [reply wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/BaseFactoryParameterResolver.java:67]: This is Che specific factory API version constant. `CURRENT_VERSION = "4.0"`

File: wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/BaseFactoryParameterResolver.java (hunk 2 above)
Comment (vinokurig) [reply wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/BaseFactoryParameterResolver.java]: done

**Review by ibuziuk — APPROVED** [review]

**Comment by artaleks9** [comment]
Verified on Eclipse Che with ` quay.io/eclipse/che-server:pr-702` - **the functionality works properly**

**Review by artaleks9 — APPROVED** [review]

**Pull request merged**